engine/src/engine.rs

```diff
@@ -1023,7 +1023,7 @@ pub fn compute_block_hash(chain_id: [u8; 32], block_height: u64, account_id: &[u
 }
 
 #[must_use]
-pub fn get_authorizer<I: IO>(io: &I) -> EngineAuthorizer {
+pub fn get_authorizer<I: IO + Copy>(io: &I) -> EngineAuthorizer {
```

Comment:
Out of curiosity, why did you add the `Copy` bounding for the `I` type?

Reply:
`Copy` is needed now because I am cheating on Rust mutability in the `get_state` function. In the case where a migration is needed the `get_state` function mutates the state, but since this happens exactly once I figured it would be overkill to change the type signature to `&mut I` and instead I copy `& I` into a mutable instance to pass to `set_state`. I know this bad practice in general, but in this case I think it is acceptable. Though if you feel otherwise then I am willing to change it.

Reply:
Got it. Thanks.